contracts/stellar-interchain-token-service/src/tests/deploy_remote_canonical_token.rs

```diff
@@ -92,13 +96,55 @@ fn deploy_remote_canonical_token_succeeds() {
     let deploy_remote_canonical_token_auth = auth_invocation!(
         &env,
         spender,
-        client.deploy_remote_canonical_token(token_address, destination_chain, spender, gas_token),
+        client.deploy_remote_canonical_token(
+            token_address,
+            destination_chain,
+            spender,
+            Some(gas_token)
+        ),
         gas_service_auth
     );
 
     assert_eq!(env.auths(), deploy_remote_canonical_token_auth);
 }
 
+#[test]
+fn deploy_remote_canonical_token_succeeds_without_gas_token() {
+    let (env, client, _, _, _) = setup_env();
+    let spender = Address::generate(&env);
+    let gas_token: Option<Token> = None;
+    let token_address = client.native_token_address();
+    let destination_chain = String::from_str(&env, "ethereum");
+
+    client.register_canonical_token(&token_address);
+
+    client
+        .mock_all_auths()
+        .set_trusted_chain(&destination_chain);
+
+    client.mock_all_auths().deploy_remote_canonical_token(
+        &token_address,
+        &destination_chain,
+        &spender,
+        &gas_token,
+    );
+
+    goldie::assert!(events::fmt_emitted_event_at_idx::<
+        InterchainTokenDeploymentStartedEvent,
+    >(
+        &env,
+        INTERCHAIN_TOKEN_DEPLOYED_WITHOUT_GAS_TOKEN_EVENT_IDX
+    ));
+
+    let deploy_remote_canonical_token_auth = auth_invocation!(
+        &env,
+        spender,
+        client.deploy_remote_canonical_token(token_address, destination_chain, spender, gas_token)
```

Comment:
```suggestion
        spender,
        client.deploy_remote_canonical_token(token_address, destination_chain, spender, gas_token)
```
`env` can be retrieved via `client.env` like for other macros. Create another PR to simplify it

Reply:
yeah, make sense. created a ticket: https://axelarnetwork.atlassian.net/browse/AXE-7831